### PR TITLE
Make Context private

### PR DESCRIPTION
**Pull request by kkaefer**
As noted in https://github.com/mapbox/mapbox-gl-native/issues/8148, some SDKs use the GL Context object. We should expose those routines and keep `gl::Context` private.

### CODE REVIEW

**Comment by kkaefer** [comment]
Contains https://github.com/mapbox/mapbox-gl-native/pull/8210 and https://github.com/mapbox/mapbox-gl-native/pull/8147

File: platform/android/config.cmake [hunk 1]
```diff
@@ -226,77 +103,182 @@ endmacro()
...
+target_compile_options(mbgl-android
     PRIVATE -fvisibility=hidden
     PRIVATE -ffunction-sections
     PRIVATE -fdata-sections
 )
 
-target_link_libraries(mapbox-gl
+target_link_libraries(mbgl-android
     PUBLIC mbgl-core
-    PUBLIC -Wl,--gc-sections
```
Comment (jfirebaugh) [inline platform/android/config.cmake]: Do we not need the `--gc-sections` option anymore?

**Comment by jfirebaugh** [comment]
Build failure should be fixed by rebasing after 83cef9eb686991d5ba31f89e3fe11cafcb4fd5bf.

File: platform/android/config.cmake (hunk 1 above)
Comment (kkaefer) [inline platform/android/config.cmake]: This is the static archive, it's still added to the .so file.

**Pull request merged**